src/Perl6/Metamodel/ParametricRoleGroupHOW.nqp

```diff
@@ -90,6 +90,20 @@ class Perl6::Metamodel::ParametricRoleGroupHOW
         $curried
     }
 
+    method bind($obj, $curried) {
+        my $candidate;
+        my %named_args := $curried.HOW.role_named_arguments($curried);
+        my @pos_args := [$obj];
+        nqp::splice(@pos_args, $curried.HOW.role_arguments($curried), +@pos_args, 0);
```

Comment:
BTW, why do you use `nqp::splice` instead of `nqp::push`?

Reply:
Splicing at the end of an array like I do here seems to run faster than pushing from a `for`. Elsewhere, there's usually a need to map over the arguments, in which case I'd expect the loop to be more efficient instead.

Reply:
Makes sense. I forgot that `push` is for single items only.